### ISSUE TITLE
Set Theory / Relational operators on a Datastream

### ISSUE THREAD

**Issue by hannesmiller** [51zero/eel-sdk#310]
Set Theory / Relational operators on a Datastream:

- union, intersection, subtract, distinct, Cartesian, zip 

All very useful performing transformations.

Hopefully the above is self explanatory but here's a link for any help/ideas: 

http://training.databricks.com/visualapi.pdf 

The thing to think about is the RHS set should already be materialised, and broadcasted to all EEL IO workers.

Spark for example has a broadcast function which can be used with *join* on a DataFrame:

```scala
df.join(broadcast(otherDf), ...)
```

**Comment by sksamuel** [comment]
Yes good idea, it's along the lines of the `join` operation that was added in 1.2.

**Comment by sksamuel** [comment]
We already have union - we have both join, concat, and union operations.
I've added Cartesian, that seems really useful. I've also added subtract and intersection.
I'm not sure what what the others really mean in the context of eel. For example, what would zip do. And unique would mean having to buffer the entire lot, so goes against how a datastream works.
So I'm happy that with join, concat, union, intersection, cartesian, subtract we've got a good set!
